pom.xml

```diff
@@ -126,6 +127,17 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
```

Comment:
Why?

Reply:
So that the plugin's version number gets saved to the manifest file, and can be accessed using `getImplementationVersion()`, for use with the version compatibility check API.

Another alternative I can think of would be to use Maven's resource filtering to inject the version number into a properties file in the project resources.

Reply:
Got it, thx. LGTM.